File: VB/ConsoleApplication1/Program.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports ClassLibrary1
Imports DevExpress.Xpo
Imports DevExpress.Xpo.DB


Namespace ConsoleApplication1
	Friend Class Program
		Private Shared ObjectLayer As IObjectLayer
		Shared Sub Main()
			Dim serializableObjectLayer As ISerializableObjectLayer = New SerializableObjectLayerServiceClient("BasicHttpBinding_ObjectLayer")
			serializableObjectLayer.CanLoadCollectionObjects.ToString()
			XpoDefault.DataLayer = Nothing
			XpoDefault.Session = Nothing
			ObjectLayer = New SerializableObjectLayerClient(serializableObjectLayer)
			Using uow As New UnitOfWork(ObjectLayer)
				Using customers As New XPCollection(Of Customer)(uow)
					For Each customer As Customer In customers
						Console.WriteLine("Company Name = {0}; ContactName = {1}", customer.CompanyName, customer.ContactName)
					Next customer
				End Using
			End Using
			Console.WriteLine("Press any key...")
			Console.ReadKey()
		End Sub
	End Class
End Namespace

File: CS/ConsoleApplication1/Program.cs
using System;
using System.Collections.Generic;
using System.Linq;
using ClassLibrary1;
using DevExpress.Xpo;
using DevExpress.Xpo.DB;


namespace ConsoleApplication1 {
    class Program {
        static IObjectLayer ObjectLayer;
        static void Main() {
            ISerializableObjectLayer serializableObjectLayer =
                new SerializableObjectLayerServiceClient("BasicHttpBinding_ObjectLayer");
            serializableObjectLayer.CanLoadCollectionObjects.ToString();
            XpoDefault.DataLayer = null;
            XpoDefault.Session = null;
            ObjectLayer = new SerializableObjectLayerClient(serializableObjectLayer);
            using(UnitOfWork uow = new UnitOfWork(ObjectLayer)) {
                using(XPCollection<Customer> customers = new XPCollection<Customer>(uow)) {
                    foreach(Customer customer in customers) {
                        Console.WriteLine("Company Name = {0}; ContactName = {1}",
                               customer.CompanyName, customer.ContactName);
                    }
                }
            }
            Console.WriteLine("Press any key...");
            Console.ReadKey();
        }
    }
}
